### PR TITLE
fix(core): get package published and working [ENG-28354]

### PR DESCRIPTION
**Pull request by devndevs**
Needed to set up forked repo to be usable by our component library 

Confirmed that the forked package is working.
image

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+const path = require('path');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer');
+const TerserJSPlugin = require('terser-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const OptimizeCSSAssetsPlugin = require('optimize-css-assets-webpack-plugin');
+const autoprefixer = require('autoprefixer');
+
```
Comment (allieholcombe) [inline webpack.config.js]: You should also be able to remove this file!

**Review by allieholcombe — DISMISSED** [review]
🚀

File: package.json [hunk 2]
```diff
@@ -18,7 +18,16 @@
...
-  "repository": "https://github.com/jbetancur/react-data-table-component",
+  "publishable": true,
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/drata/react-data-table-component"
+  },
+  "publishConfig": {
+    "registry": "https://npm.pkg.github.com",
+    "access": "restricted",
+    "token": "${NPM_TOKEN}"
```
Comment (TylerLHenderson) [inline package.json:29]: is this line doing anything? I don't see it in the config
https://docs.npmjs.com/cli/v9/using-npm/config

File: tsconfig.json [hunk 3]
```diff
@@ -5,31 +5,27 @@
 		"experimentalDecorators": true,
 		"forceConsistentCasingInFileNames": true,
 		"isolatedModules": true,
-		"noEmit": true,
+		"noEmit": false,
```
Comment (TylerLHenderson) [inline tsconfig.json]: with this off we have to use another tool to compile the code, is the package working in web with `noEmit=false`?
https://www.typescriptlang.org/tsconfig#noEmit

File: tsconfig.json [hunk 4]
```diff
@@ -5,31 +5,27 @@
...
-			"dom"
-		],
-		"declaration": true
+		"lib": ["es2018", "dom"],
 	},
 	"include": [
-		"src/**/*",
-		"stories/**/*"
-	],
+		"./src"],
```
Comment (TylerLHenderson) [inline tsconfig.json]:
```suggestion
	"include": ["./src"],
```

File: package.json (hunk 2 above)
Comment (devndevs) [reply package.json:29]: This is what I needed to add to publish the package. I used this https://www.notion.so/drata/How-to-publish-a-platform-package-06a0e71b9af04b8f90b2a418787a13ae?pvs=4

**Review by TylerLHenderson — APPROVED** [review]

**Review by allieholcombe — APPROVED** [review]

**Pull request merged**